nerfstudio/viewer/viewer_elements.py

```diff
@@ -710,3 +715,96 @@ def _create_gui_handle(self, viser_server: ViserServer) -> None:
         self.gui_handle = viser_server.add_gui_vector3(
             self.name, self.default_value, step=self.step, disabled=self.disabled, visible=self.visible, hint=self.hint
         )
+
+
+class ViewerPlot(ViewerElement[go.Figure]):
+    """Base class for viewer figures, using plotly backend.
+    Includes misc wrapper methods for setting plotly figure properties.
+    """
+    gui_handle: GuiPlotlyHandle
+
+    _figure: go.Figure
+    """Figure to be displayed. Do not access this directly, exists only for initial statekeeping."""
+    _aspect: float
+    """Aspect ratio of the plot (h/w). Default is 1.0."""
+
+    def __init__(
+        self,
+        figure: Optional[go.Figure] = None,
+        aspect: float = 1.0,
+        visible: bool = True,
+    ):
+        """
+        Args:
+        - figure: The plotly figure to display -- if None, an empty figure is created.
+        - aspect: Aspect ratio of the plot (h/w). Default is 1.0.
+        - visible: If the plot is visible.
+        """
+        self._figure = go.Figure() if figure is None else figure
+        self._aspect = aspect
+        super().__init__(name="", visible=visible)  # plots have no name.
+
+    def _create_gui_handle(self, viser_server: ViserServer) -> None:
+        self.gui_handle = viser_server.add_gui_plotly(
+            figure=self._figure, visible=self.visible, aspect=self._aspect
+        )
+
+    def install(self, viser_server: ViserServer) -> None:
+        self._create_gui_handle(viser_server)
+        assert self.gui_handle is not None
+
+    @property
+    def figure(self):
+        assert self.gui_handle is not None
+        return self.gui_handle.figure
+
+    @figure.setter
+    def figure(self, figure: go.Figure):
```

Comment:
it would be nice to automatically apply the dark mode+aspect to new figures as well, so that the user doesn't have to call dark_mode and aspect every time they update/change something

Reply:
`aspect` doesn't need to be set for new figures, but you're right about `dark_mode`. :+1:

Reply:
Should be addr'ed in https://github.com/nerfstudio-project/nerfstudio/pull/3129/commits/ca04cbc0d9e44a2dba41fe18f23d5d428071cb43.